scripts/west_commands/runners/__init__.py

```diff
@@ -2,34 +2,108 @@
+                    'dependencies properly. Please report this '
+                    'to the zephyr developers.')
+
 # We import these here to ensure the ZephyrBinaryRunner subclasses are
 # defined; otherwise, ZephyrBinaryRunner.get_runners() won't work.
+#
+# Runners are supposed to gracefully handle failures when they import
+# anything outside of stdlib, but they sometimes do not. Catch ImportError
+# to handle this.
 
 # Explicitly silence the unused import warning.
 # flake8: noqa: F401
 # Keep this list sorted by runner name.
-from runners import blackmagicprobe
-from runners import bossac
-from runners import canopen_program
-from runners import dediprog
-from runners import dfu
-from runners import esp32
-from runners import hifive1
-from runners import intel_s1000
-from runners import jlink
-from runners import misc
-from runners import nios2
-from runners import nrfjprog
-from runners import nsim
-from runners import openocd
-from runners import pyocd
-from runners import qemu
-from runners import stm32flash
-from runners import xtensa
-from runners import mdb
-from runners import stm32cubeprogrammer
+try:
+    from runners import blackmagicprobe
+except ImportError:
+    _log_bad_import('blackmagicprobe')
+try:
+    from runners import bossac
+except ImportError:
+    _log_bad_import('bossac')
+try:
+    from runners import canopen_program
+except ImportError:
+    _log_bad_import('canopen_program')
+try:
+    from runners import dediprog
+except ImportError:
+    _log_bad_import('dediprog')
+try:
+    from runners import dfu
+except ImportError:
+    _log_bad_import('dfu')
+try:
+    from runners import esp32
+except ImportError:
+    _log_bad_import('esp32')
+try:
+    from runners import hifive1
+except ImportError:
+    _log_bad_import('hifive1')
+try:
+    from runners import intel_s1000
+except ImportError:
+    _log_bad_import('intel_s1000')
+try:
+    from runners import jlink
+except ImportError:
+    _log_bad_import('jlink')
+try:
+    from runners import misc
+except ImportError:
+    _log_bad_import('misc')
+try:
+    from runners import nios2
+except ImportError:
+    _log_bad_import('nios2')
+try:
+    from runners import nrfjprog
+except ImportError:
+    _log_bad_import('nrfjprog')
+try:
+    from runners import nsim
+except ImportError:
+    _log_bad_import('nsim')
+try:
+    from runners import openocd
+except ImportError:
+    _log_bad_import('openocd')
+try:
+    from runners import pyocd
+except ImportError:
+    _log_bad_import('pyocd')
+try:
+    from runners import qemu
+except ImportError:
+    _log_bad_import('qemu')
+try:
+    from runners import stm32flash
+except ImportError:
+    _log_bad_import('stm32flash')
+try:
+    from runners import xtensa
+except ImportError:
+    _log_bad_import('xtensa')
+try:
+    from runners import mdb
+except ImportError:
+    _log_bad_import('mdb')
+try:
+    from runners import stm32cubeprogrammer
+except ImportError:
+    _log_bad_import('stm32cubeprogrammer')
```

Comment:
Have you considered `importlib.import_module()` ? I've never used it but this repetition seems begging for it.

Reply:
> Have you considered `importlib.import_module()` ? I've never used it but this repetition seems begging for it.

Yes, I considered it. In general I think importlib is heavy machinery and I try to avoid using it when a simple `import`  statement will do. I can make the change if you'd like, though.

Reply:
I don't like the sound of "heavy machinery" either but I don't know what it would mean in this particular context. If it would make the code much shorter then I suggest giving it at least a time-limited try.

Reply:
> I don't like the sound of "heavy machinery" either but I don't know what it would mean in this particular context.

It's more magical and harder to read IMO. But sure, I'll use it instead.

Reply:
> If it would make the code much shorter then I suggest giving it at least a time-limited try.

Done.

Reply:
You know, now that we've gone this route, at some point it may make sense to change the `runners` package name to `zephyr_runners`, make it a PEP 420 namespace package, and make `get_runner_cls` just use importlib directly to open up the possibility for user-defined runners. I'll file it in the back of my head somewhere.